### PR TITLE
spelling fixes

### PR DESCRIPTION
**Pull request by ka7**
patch contains some spelling fixes ( most in /comments/, some in debug/error-msg in tests ) as found by a bot ( https://github.com/ka7/misspell_fixer )

### Description

* spelling fixes
* one script-fix in tests/zfs-tests/tests/functional/nopwrite/nopwrite_volume.ksh (typo in varialbe name)

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the ZFS on Linux code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

File: tests/zfs-tests/tests/functional/refreserv/refreserv_005_pos.ksh [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #
 # STRATEGY:
 #	1. Create volume on filesystem
-#	2. Setting quota for parenet filesytem
+#	2. Setting quota for parenet filesystem
```
Comment (inkdot7) [inline tests/zfs-tests/tests/functional/refreserv/refreserv_005_pos.ksh]: parent

File: module/zfs/spa_config.c [hunk 2]
```diff
@@ -222,7 +222,7 @@ spa_config_write(spa_config_dirent_t *dp, nvlist_t *nvl)
  * the configuration has been synced to the MOS. This exposes a window where
  * the MOS config will have been updated but the cache file has not. If
  * the system were to crash at that instant then the cached config may not
- * contain the correct information to open the pool and an explicity import
+ * contain the correct information to open the pool and an explicitly import
```
Comment (inkdot7) [inline module/zfs/spa_config.c]: explicit

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_003_pos.ksh [hunk 3]
```diff
@@ -129,25 +129,25 @@ for clone in $child_fs1_snap_clone $child_fs1_snap_clone1; do
 done
 
 log_note "Verify that 'zfs destroy -r' fails to destroy dataset " \
-	"with clone dependant outside it."
+	"with clone dependent outside it."
```
Comment (inkdot7) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_003_pos.ksh]: Just noting that the changes dependant -> dependent in this file seems to be BrE -> AmE changes.
Would it be better with 'dependent clone' than 'clone dependent'?

File: module/zfs/rrwlock.c [hunk 4]
```diff
@@ -313,8 +313,8 @@ rrw_tsd_destroy(void *arg)
  * The idea is to split single busy lock into array of locks, so that
  * each reader can lock only one of them for read, depending on result
  * of simple hash function.  That proportionally reduces lock congestion.
- * Writer same time has to sequentially aquire write on all the locks.
- * That makes write aquisition proportionally slower, but in places where
+ * Writer same time has to sequentially acquire write on all the locks.
```
Comment (inkdot7) [inline module/zfs/rrwlock.c]: 'Writer same' -> 'Writer at the same' ?

File: tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_003_pos.ksh (hunk 3 above)
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_003_pos.ksh]: Yes, 'dependent clone' is better let's flip this.

**Comment by behlendorf** [comment]
Squashed and merged.

**Pull request merged**